Scripts/Runtime/Entities/MemsetBufferJob.cs

```diff
@@ -13,6 +16,36 @@ namespace Anvil.Unity.DOTS.Entities
     [BurstCompile]
     public struct MemsetBufferJob<T> : IJobParallelForBatch where T : struct, IBufferElementData
     {
+        // The minimum number of elements to process in a single batch.
+        // Attempts to keep batch sizes large enough to prevent multiple cores from copying the same
+        // data into cache. This job's operation is so simple the copy into cache is probably the most expensive part.
+        //
+        // Ex: We shouldn't split a set of 10 across 5 threads.
+        //
+        // Re CacheSize: If generalizing to a 64kb cache size is good enough for Unity it's good 
+        // enough for us.
+        private static readonly float MIN_BATCH_SIZE_PER_THREAD = math.max(1f, JobsUtility.CacheLineSize * 1024 / (float)UnsafeUtility.SizeOf<T>());
+
+        /// <summary>
+        /// Calculate an ideal batch size per thread.
+        /// Aim to spread work across as many threads as possible while satisfying (in order of importance):
+        ///  - Aligning batch size to cache line size
+        ///  - Keeping batch sizes large enough to overcome the cost of splitting.
+        ///  - Minimizing the number of batches (there's overhead in each batch run)
+        /// </summary>
+        /// <remarks>
+        /// This approach minimizes frame time consumed not total computation time. 
+        /// Single thread will always consume the fewest CPU cycles.
+        /// </remarks>
+        /// <param name="length">The total length of the data set</param>
+        /// <returns>The ideal batch size</returns>
+        public static int CalculateOptimalBatchSize(int length)
+        {
+            float maxBatches = length / MIN_BATCH_SIZE_PER_THREAD;
+            // Spread the max batches across all threads but round up to the nearest MIN_BATCH_SIZE_PER_THREAD.
+            return (int)(math.ceil(maxBatches / (float)JobsUtility.JobWorkerCount) * MIN_BATCH_SIZE_PER_THREAD);
```

Comment:
I ran into this just recently, but you might want to use `JobsUtility.JobWorkerCount + 1`

The value returned will give you the number of worker threads that you have (9 on your M1 system, 15 on my older intel 8 core) but you also have the main thread which jobs can still be scheduled on. 

Looking further at the actual code for that getter, I was surprised to see that it is a setter too... which seems dangerous? 

But apparently there are some use cases for it albeit much older: https://www.gamedev.net/tutorials/programming/general-and-gameplay-programming/unity-job-system-excessive-multithreading-might-hurt-your-performance-r5258/